File: app/src/main/java/org/mifos/mobilebanking/models/notification/MifosNotification.java
package org.mifos.mobilebanking.models.notification;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

import org.mifos.mobilebanking.api.local.SelfServiceDatabase;

/**
 * Created by dilpreet on 13/9/17.
 */
@Table(database = SelfServiceDatabase.class)
public class MifosNotification extends BaseModel {

    @PrimaryKey
    Long timeStamp;

    @Column
    String msg;

    @Column
    Boolean read;

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public long getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(long timeStamp) {
        this.timeStamp = timeStamp;
    }

    public boolean isRead() {
        return read;
    }

    public void setRead(boolean read) {
        this.read = read;
    }
}

File: app/src/main/java/org/mifos/mobile/models/notification/MifosNotification.kt
package org.mifos.mobile.models.notification

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.structure.BaseModel
import org.mifos.mobile.api.local.SelfServiceDatabase

/**
 * Created by dilpreet on 13/9/17.
 */
@Table(database = SelfServiceDatabase::class)
class MifosNotification : BaseModel() {

    @JvmField
    @PrimaryKey
    var timeStamp: Long = 0

    @JvmField
    @Column
    var msg: String? = null

    @JvmField
    @Column
    var read: Boolean? = null
    fun getTimeStamp(): Long {
        return timeStamp
    }

    fun setTimeStamp(timeStamp: Long) {
        this.timeStamp = timeStamp
    }

    fun isRead(): Boolean {
        return read ?: false
    }

    fun setRead(read: Boolean?) {
        this.read = read
    }
}
